File: combat/src/classicthunder/combat/character/CharacterStats.java
package classicthunder.combat.character;

public class CharacterStats {

    private final CharacterType characterType;

    private int health;
    private int block;

    public CharacterStats(CharacterType characterType, int startingHealth) {

        this.characterType = characterType;
        this.health = startingHealth;
        this.block = 0;
    }

    public void AdjustHealth(int delta) {
        this.health += delta;
    }

    public int GetHealth() {
        return this.health;
    }

    public void AdjustBlock(int delta) {
        this.block += delta;
    }

    public int GetBlock() {
        return this.block;
    }

    public CharacterType getCharacterType() {
        return characterType;
    }
}

File: combat/src/classicthunder/components/character/CharacterStats.kt
package classicthunder.components.character

class CharacterStats(
    val characterType: CharacterType,
    private var health: Int,
    private var block: Int = 0)
{
    fun AdjustHealth(delta: Int) {
        health += delta
    }

    fun GetHealth(): Int {
        return health
    }

    fun AdjustBlock(delta: Int) {
        block += delta
    }

    fun GetBlock(): Int {
        return block
    }
}